dist/src/main/java/io/camunda/zeebe/shared/management/JobStreamEndpoint.java

```diff
@@ -0,0 +1,136 @@
+ * refers to {@link io.camunda.zeebe.transport.stream.api.RemoteStream} instances registered on the
+ * broker, and {@code client} refers to all {@link ClientStream} instances registered on the
+ * gateway.
+ *
+ * <p>If running a broker with an embedded gateway, both can be viewed; otherwise, one or the other
+ * will always be empty.
+ */
+@Component
+@WebEndpoint(id = "jobstreams")
+public final class JobStreamEndpoint {
+  private static final Set<String> TYPES = Set.of("remote", "client");
+
+  private final JobStreamEndpointService service;
+  private final AtomixCluster cluster;
+
+  @Autowired
+  public JobStreamEndpoint(final JobStreamEndpointService service, final AtomixCluster cluster) {
+    this.service = service;
+    this.cluster = cluster;
+  }
+
+  @ReadOperation
+  public WebEndpointResponse<JobStreams> list() {
+    return new WebEndpointResponse<>(
+        new JobStreams(getRemoteStreams(), getClientStreams()),
+        200,
+        MimeTypeUtils.APPLICATION_JSON);
+  }
+
+  @ReadOperation
+  public WebEndpointResponse<?> list(final @Selector String type) {
+    if (!TYPES.contains(type)) {
+      return new WebEndpointResponse<>(
+          Map.of("error", "No known stream type '%s'; should be one of %s".formatted(type, TYPES)),
+          400,
+          MimeTypeUtils.APPLICATION_JSON);
+    }
+
+    final Collection<?> streams = "client".equals(type) ? getClientStreams() : getRemoteStreams();
+    return new WebEndpointResponse<>(streams, 200, MimeTypeUtils.APPLICATION_JSON);
+  }
+
+  private Collection<RemoteJobStream> getRemoteStreams() {
+    return transformRemote(service.remoteJobStreams());
+  }
+
+  private Collection<ClientJobStream> getClientStreams() {
+    return transformClient(service.clientJobStreams());
+  }
+
+  private Collection<RemoteJobStream> transformRemote(
+      final Collection<RemoteStreamInfo<JobActivationProperties>> streams) {
+    return streams.stream().map(this::transformRemote).toList();
+  }
+
+  private Collection<ClientJobStream> transformClient(
+      final Collection<ClientStream<JobActivationProperties>> streams) {
+    return streams.stream().map(this::transformClient).toList();
+  }
+
+  private RemoteJobStream transformRemote(final RemoteStreamInfo<JobActivationProperties> stream) {
+    final var consumers =
+        stream.consumers().stream()
+            .map(id -> new RemoteStreamId(id.streamId(), id.receiver().id()))
+            .toList();
+    return new RemoteJobStream(
+        BufferUtil.bufferAsString(stream.streamType()), transform(stream.metadata()), consumers);
+  }
+
+  private ClientJobStream transformClient(final ClientStream<JobActivationProperties> stream) {
+    final var brokers =
+        cluster.getMembershipService().getMembers().stream()
+            .map(Member::id)
+            .filter(stream::isConnected)
+            .map(MemberId::id)
+            .map(Integer::valueOf)
```

Comment:
I don't think it's worth handling here if the broker ID is not an integer, but we can do it. We could also leave it as a string, but I figured it will be nicer to assert things around brokers if we can use the actual ID.